src/rust/types/src/order.rs

```diff
@@ -156,6 +190,72 @@ pub enum OrderState {
     Evicted,
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "snake_case")
+)]
+#[repr(u8)]
+pub enum CancelType {
+    Both,
+    Maker,
+    Taker,
+}
+
+impl TryFrom<u8> for CancelType {
+    type Error = TypeError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::Both),
+            1 => Ok(Self::Maker),
+            2 => Ok(Self::Taker),
+            _ => Err(TypeError::ConversionError {
+                name: "CancelType".to_string(),
+            }),
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "snake_case")
+)]
+#[repr(u8)]
+pub enum CancelReason {
+    Eviction,
+    ImmediateOrCancel,
+    ManualCancel,
+    MaxQuoteTraded,
+    NotEnoughLiquidity,
+    SelfMatchMaker,
+    SelfMatchTaker,
+    TooSmallAfterMatching,
+}
+
+impl TryFrom<u8> for CancelReason {
+    type Error = TypeError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(CancelReason::Eviction),
+            2 => Ok(CancelReason::ImmediateOrCancel),
+            3 => Ok(CancelReason::ManualCancel),
+            4 => Ok(CancelReason::MaxQuoteTraded),
+            5 => Ok(CancelReason::NotEnoughLiquidity),
+            6 => Ok(CancelReason::SelfMatchMaker),
+            7 => Ok(CancelReason::SelfMatchTaker),
+            8 => Ok(CancelReason::TooSmallAfterMatching),
```

Comment:
The last few reasons are out of date, it should be:

* 8: TooSmallToFillLot
* 9: ViolatedLimitPrice

Reply:
Seems that this is duplicated, and this type is already in another place. Went away and removed it all.